src/main/java/com/fasterxml/jackson/databind/AnnotationIntrospector.java

```diff
@@ -967,6 +967,23 @@ public PropertyName findNameForSerialization(Annotated a) {
         return null;
     }
 
+    /**
+     * Method for checking whether given method has an annotation
+     * that suggests the return value of annotated method
+     * should be used as "the key" of the object instance; usually
+     * serialized as a primitive value such as String or number.
+     *
+     * @return {@link Boolean#TRUE} if such annotation is found and is not disabled;
+     *   {@link Boolean#FALSE} if disabled annotation (block) is found (to indicate
+     *   accessor is definitely NOT to be used "as value"); or `null` if no
+     *   information found.
+     *
+     * @since TODO
+     */
+    public Boolean hasAsKey(Annotated a) {
```

Comment:
One suggestion: could you take `MapperConfig<?>` as the first parameter? While it is not currently needed, it is something I have had to retrofit for a few methods (see f.ex `findCreatorAnnotation()`), and in 3.0 every method will be changed to take it. So adding it now reduces conversion work slightly.